pkg/gatherers/common/gather_logs.go

```diff
@@ -112,6 +115,12 @@ func CollectLogsFromContainers( //nolint:gocyclo
 				tailLines = nil
 			}
 
+			if containersFilter.MaxNamespaceContainers > 0 && len(records) >= containersFilter.MaxNamespaceContainers {
+				errs = append(errs, fmt.Errorf("skipping %s for %s. Max containers per namespace reached (max: %d)",
```

Comment:
Isn't it gonna spam the errors? I think you can exit the loop here

Reply:
Yep, it will. But, at the same time, we want to know about the containers that we are skipping. So, another option would be to have a single error with the number of pods skipped. What do you think?

Reply:
Single error with the number of skipped pods is much better

Reply:
Done.